zerocopy-derive/src/lib.rs

```diff
@@ -908,15 +846,28 @@ const STRUCT_UNION_INTO_BYTES_CFG: Config<StructRepr> = Config {
 // An enum is `IntoBytes` if it is field-less and has a defined repr.
 
 fn derive_into_bytes_enum(ast: &DeriveInput, enm: &DataEnum) -> proc_macro2::TokenStream {
```

Comment:
Presently, this derive:
```rust
#[derive(IntoBytes)]
#[repr(u8)]
enum Foo {
    A(u8),
    B(u8),
}
```
...expands to:
```rust
const _: () = {
    #[repr(u8)]
    #[allow(dead_code)]
    enum ___ZerocopyTag {
        A,
        B,
    }
    #[allow(deprecated)]
    unsafe impl ::zerocopy::IntoBytes for Foo
    where
        u8: ::zerocopy::IntoBytes,
        u8: ::zerocopy::IntoBytes,
        ::zerocopy::macro_util::HasPadding<
            Foo,
            {
                false
                    || ::zerocopy::macro_util::core_reexport::mem::size_of::<Foo>()
                        != (::zerocopy::macro_util::core_reexport::mem::size_of::<
                            ___ZerocopyTag,
                        >()
                            + ::zerocopy::macro_util::core_reexport::mem::size_of::<
                                u8,
                            >())
                    || ::zerocopy::macro_util::core_reexport::mem::size_of::<Foo>()
                        != (::zerocopy::macro_util::core_reexport::mem::size_of::<
                            ___ZerocopyTag,
                        >()
                            + ::zerocopy::macro_util::core_reexport::mem::size_of::<
                                u8,
                            >())
            },
        >: ::zerocopy::macro_util::ShouldBe<false>,
    {
        fn only_derive_is_allowed_to_implement_this_trait() {}
    }
};
```

I think this is the only place we're generating items outside of the trait impl. Can we move that generated `___ZerocopyTag` type into the body of that const expr? If we can do so, then we don't need to generate the wrapping `const _: () = {...}` blocks.

Reply:
Done. Note that the stderr output for `enum.rs` has changed as a result.